provider/core/src/fork/by_key.rs

```diff
@@ -128,6 +128,23 @@ impl<P0: BufferProvider, P1: BufferProvider> BufferProvider for ForkByKeyProvide
     }
 }
 
+impl<P0: AnyProvider, P1: AnyProvider> AnyProvider for ForkByKeyProvider<P0, P1> {
+    #[inline]
+    fn load_any(&self, req: &DataRequest) -> Result<AnyResponse, DataError> {
+        let result = self.0.load_any(req);
+        if !matches!(
+            result,
```

Comment:
thought: might we worth having a convenient `.is_resource_key()` on DataError though that doesn't help unwrapping the result. Perhaps `DataError::is_resource_key(err: Result<_, Self>) -> bool`.